op-deployer/pkg/deployer/state/intent.go

```diff
@@ -32,35 +33,313 @@ func (d DeploymentStrategy) Check() error {
+		return errors.New("must define at least one l2 chain")
+	}
 
-	Chains []*ChainIntent `json:"chains" toml:"chains"`
+	for _, chain := range c.Chains {
+		if err := chain.CheckNoZeroValues(); err != nil {
+			return err
+		}
+	}
 
-	GlobalDeployOverrides map[string]any `json:"globalDeployOverrides" toml:"globalDeployOverrides"`
+	return nil
 }
 
-func (c *Intent) L1ChainIDBig() *big.Int {
-	return big.NewInt(int64(c.L1ChainID))
+func (c *Intent) validateStrictConfig() error {
+	if err := c.validateStandardValues(); err != nil {
+		return err
+	}
+
+	challenger, _ := standard.ChallengerAddressFor(c.L1ChainID)
+	l1ProxyAdminOwner, _ := standard.L1ProxyAdminOwner(c.L1ChainID)
+	for chainIndex := range c.Chains {
+		if c.Chains[chainIndex].Roles.Challenger != challenger {
+			return fmt.Errorf("invalid challenger address for chain: %s", c.Chains[chainIndex].ID)
+		}
+		if c.Chains[chainIndex].Roles.L1ProxyAdminOwner != l1ProxyAdminOwner {
+			return fmt.Errorf("invalid l1ProxyAdminOwner address for chain: %s", c.Chains[chainIndex].ID)
+		}
+	}
+
+	return nil
+}
+
+// Ensures the following:
+//  1. no zero-values for non-standard fields (user should have populated these)
+//  2. no non-standard values for standard fields (user should not have changed these)
+func (c *Intent) validateStandardValues() error {
+	standardSuperchainRoles, err := getStandardSuperchainRoles(c.L1ChainID)
+	if err != nil {
+		return fmt.Errorf("error getting standard superchain roles: %w", err)
+	}
+	if *c.SuperchainRoles != *standardSuperchainRoles {
+		return fmt.Errorf("SuperchainRoles does not match standard value")
+	}
+
+	for _, chain := range c.Chains {
+		if chain.ID == emptyHash {
+			return fmt.Errorf("missing l2 chain ID")
+		}
+		if err := chain.Roles.CheckNoZeroAddresses(); err != nil {
+			return err
+		}
+		if chain.Eip1559DenominatorCanyon != standard.Eip1559DenominatorCanyon ||
+			chain.Eip1559Denominator != standard.Eip1559Denominator ||
+			chain.Eip1559Elasticity != standard.Eip1559Elasticity {
+			return fmt.Errorf("%w: chainId=%s", ErrNonStandardValue, chain.ID)
+		}
+		if chain.BaseFeeVaultRecipient == emptyAddress ||
+			chain.L1FeeVaultRecipient == emptyAddress ||
+			chain.SequencerFeeVaultRecipient == emptyAddress {
+			return fmt.Errorf("%w: chainId=%s", ErrFeeVaultZeroAddress, chain.ID)
+		}
+	}
+
+	return nil
+}
+
+func getStandardSuperchainRoles(l1ChainId uint64) (*SuperchainRoles, error) {
+	superCfg, err := standard.SuperchainFor(l1ChainId)
+	if err != nil {
+		return nil, fmt.Errorf("error getting superchain config: %w", err)
+	}
+
+	proxyAdmin, _ := standard.ManagerOwnerAddrFor(l1ChainId)
+	guardian, _ := standard.GuardianAddressFor(l1ChainId)
+
+	superchainRoles := &SuperchainRoles{
+		ProxyAdminOwner:       proxyAdmin,
+		ProtocolVersionsOwner: common.Address(*superCfg.Config.ProtocolVersionsAddr),
+		Guardian:              guardian,
+	}
+
+	return superchainRoles, nil
+}
+
+func (c *Intent) SetInitValues(l2ChainIds []common.Hash) error {
```

Comment:
Rather than doing this on the `Intent`'s implementation, can we make a set of initializer methods? For example, `NewIntentCustom`, `NewIntentStrict`, etc. This leaves calling the right initializer up to the caller, and simplifies the code here.